snapcraft/_options.py

```diff
@@ -150,6 +150,10 @@ def is_cross_compiling(self):
         return self.__target_machine != self.__platform_arch
 
     @property
+    def target_arch_specified(self):
+        return self.__target_arch_specified
```

Comment:
Looking at this from an external point of view, this seems a slightly odd property to make part of the class's public API. It begs the question: "Target arch specified for what?" Why don't we just save the target arch? Then if it's `None` we can start making assumptions about it.